pkg/azuredisk/controllerserver.go

```diff
@@ -293,6 +299,14 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 	contentSource := &csi.VolumeContentSource{}
 	for k, v := range customTagsMap {
 		tags[k] = v
+		if k == userAgentField {
```

Comment:
user agent does not need to be set in storage class, we only need to hard code in our driver or set as driver parameter in driver start up

Reply:
@andyzhangx We need a way to track Azure cloud operations initiated by partner ISVs on behalf of another customer. This is done through the user agent strings as described in [Azure customer usage attribution](https://docs.microsoft.com/en-us/azure/marketplace/azure-partner-customer-usage-attribution).

Reply:
I would suggest add a new parameter(`userAgent`) instead of customer tags in storage class to track csi driver operations. if not set, use hard coded userAgent in driver, the implementation in this PR is too redudunt, here is an clean example: https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/754/commits/ac18a0b4df486ad40d4479ce8d6bb2567cd03fb5

Reply:
The logic that was behind this was that ISVs would need a specific storage class created that tags useragent on operations, but wouldn't want the rest of the operations to be tagged with the Useragent. Vybava is verifying what method makes the most sense, but this was the implementation we initially decided to go for

Reply:
while current implementation would set useragent for all operations done by the driver since this whole driver would use only one useragent. Is there a customer asking for this feature? specify useragent in storage class(no matter it's tags or `userAgent` field) is not a good idea: 1) PV does not use this storage class would also get the same useragent 2) if driver restart, userAgent setting would be gone.

Reply:
I created another PR to change default userAgent(https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/969), though it's not related to this scenario.